package/kernel/mac80211/files/lib/wifi/mac80211.sh

```diff
@@ -81,13 +85,39 @@ detect_mac80211() {
 		htmode=""
 		ht_capab=""
 
-		iw phy "$dev" info | grep -q 'Capabilities:' && htmode=HT20
+		iw_phy_info=$( iw phy "${dev}" info )
+
+		echo "${iw_phy_info}" | fgrep -q 'Capabilities:' && htmode=HT20
+
+		if echo "${iw_phy_info}" | egrep -q '(49|5[0-9])[0-9][0-9] MHz' ; then
 
-		iw phy "$dev" info | grep -q '5180 MHz' && {
 			mode_band="a"
-			channel="36"
-			iw phy "$dev" info | grep -q 'VHT Capabilities' && htmode="VHT80"
-		}
+
+			channel=$( echo "${iw_phy_info}" | \
```

Comment:
There's a problem here (also with the old code - but nobody complained because of it).
You can't really have a channel at that time without having set the country (code) as well.
Problem is that the country setting affects the channel lists to a major degree and currently, it's set much later in drv_mac80211_setup().

What I think would make more sense is to tell hostapd to select a channel (for example by setting ```channel=acs_survey``` in the default configuration and improve the channel selection
there if necessary.